File: lib/services/audio_io_native.dart
import 'dart:io';
import 'dart:typed_data';
import 'package:path_provider/path_provider.dart';

Future<Uint8List?> readAudioBytes(String pathOrBlobUrl) async {
  try {
    final file = File(pathOrBlobUrl);
    if (await file.exists()) {
      return await file.readAsBytes();
    }
  } catch (_) {}
  return null;
}

Future<void> deleteAudioFile(String pathOrBlobUrl) async {
  try {
    final file = File(pathOrBlobUrl);
    if (await file.exists()) {
      await file.delete();
    }
  } catch (_) {}
}

Future<String> getTempAudioPath() async {
  final dir = await getTemporaryDirectory();
  return '${dir.path}/ser_${DateTime.now().millisecondsSinceEpoch}.wav';
}

File: lib/services/audio_io_native.js
import * as FileSystem from 'expo-file-system';

export async function readAudioBytes(pathOrBlobUrl) {
  try {
    const fileInfo = await FileSystem.getInfoAsync(pathOrBlobUrl);
    if (fileInfo.exists) {
      const base64 = await FileSystem.readAsStringAsync(pathOrBlobUrl, {
        encoding: FileSystem.EncodingType.Base64,
      });
      const binaryString = atob(base64);
      const len = binaryString.length;
      const bytes = new Uint8Array(len);
      for (let i = 0; i < len; i++) {
        bytes[i] = binaryString.charCodeAt(i);
      }
      return bytes;
    }
  } catch (_) {}
  return null;
}

export async function deleteAudioFile(pathOrBlobUrl) {
  try {
    const fileInfo = await FileSystem.getInfoAsync(pathOrBlobUrl);
    if (fileInfo.exists) {
      await FileSystem.deleteAsync(pathOrBlobUrl, { idempotent: true });
    }
  } catch (_) {}
}

export async function getTempAudioPath() {
  const dir = FileSystem.cacheDirectory;
  return `${dir}ser_${Date.now()}.wav`;
}
